lang/py/src/avro/io.py

```diff
@@ -478,6 +573,77 @@ def write_crc32(self, bytes):
     """
     self.write(STRUCT_CRC32.pack(crc32(bytes) & 0xffffffff));
 
+  def write_date_int(self, datum):
+    """
+    Encode python date object as int.
+    It stores the number of days from
+    the unix epoch, 1 January 1970 (ISO calendar).
+    """
+    delta_date = datum - datetime.date(1970, 1, 1)
+    if delta_date.days < 0:
+      raise schema.AvroException("Cannot encode date before 1 January 1970 (ISO calendar)")
+    self.write_int(delta_date.days)
+
+  def _validate_time_datum(self, value, time_type):
+    if (time_type in [constants.TIME_MILLIS] and
+        not (UNSIGNED_INT_MIN_VALUE <= value <= UNSIGNED_INT_MAX_VALUE)):
+      raise schema.AvroException("Cannot encode %s logicaltype schema with value outside (%d, %d)." % (
+        time_type,
+        UNSIGNED_INT_MIN_VALUE,
+        UNSIGNED_INT_MAX_VALUE)
+      )
+    elif (time_type in [constants.TIME_MICROS, constants.TIMESTAMP_MILLIS, constants.TIMESTAMP_MICROS] and
+        not (UNSIGNED_LONG_MIN_VALUE <= value <= UNSIGNED_LONG_MAX_VALUE)):
+      raise schema.AvroException("Cannot encode %s logicaltype schema with value outside (%d, %d)." % (
+        time_type,
+        UNSIGNED_LONG_MIN_VALUE,
+        UNSIGNED_LONG_MAX_VALUE)
+      )
+
+  def write_time_millis_int(self, datum):
+    """
+    Encode python time object as int.
+    It stores the number of milliseconds from midnight, 00:00:00.000
+    """
+    offset_milliseconds = (((datum.hour * 60) + datum.minute)*60 + datum.second)*1000 + datum.microsecond // 1000
+    self._validate_time_datum(offset_milliseconds, constants.TIME_MILLIS)
```

Comment:
i think we dont need to validate `time_millis` and `time_micros` because its not possible to create a python `datetime.time` datum with negative hr/min/sec value. 
Also all values of `datatime.time` can we encoded with `int` (millisecond precision) and with `long` (microsecond precision) 

```
>>> dat= datetime.time(-1,2,3)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: hour must be in 0..23
>>> dat= datetime.time(1,-22,3)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: minute must be in 0..59
>>> dat= datetime.time(1,2,-3)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: second must be in 0..59
>>> 
```